src/_pytest/_code/code.py

```diff
@@ -412,19 +465,15 @@ def __init__(self, tup=None, exprinfo=None):
                     exprinfo = py.io.saferepr(tup[1])
                 if exprinfo and exprinfo.startswith(self._assert_start_repr):
                     self._striptext = "AssertionError: "
-        self._excinfo = tup
-        #: the exception class
-        self.type = tup[0]
-        #: the exception instance
-        self.value = tup[1]
-        #: the exception raw traceback
-        self.tb = tup[2]
+
         #: the exception type name
         self.typename = self.type.__name__
         #: the exception traceback (_pytest._code.Traceback instance)
         self.traceback = _pytest._code.Traceback(self.tb, excinfo=ref(self))
 
     def __repr__(self):
+        if self._excinfo is None:
+            return "<ExceptionInfo for raises contextmanager>"
```

Comment:
"for raises contextmanager" -> I think this is used in other places no? If so then this is a bit misleading. Perhaps:

```suggestion
            return "<ExceptionInfo (no exception)>"
```

Reply:
currently its only used for that

before exception info objects where abused in raises they always had consistent state